tests/TestCase.php

```diff
@@ -96,27 +97,7 @@ protected static function getDatabaseName(): string
      */
     public function assertMatchesDocument($expectedDocument, $actualDocument): void
     {
-        $normalizedExpectedDocument = $this->normalizeBSON($expectedDocument);
-        $normalizedActualDocument = $this->normalizeBSON($actualDocument);
-
-        $extraKeys = [];
-
-        /* Avoid unsetting fields while we're iterating on the ArrayObject to
-         * work around https://bugs.php.net/bug.php?id=70246 */
-        foreach ($normalizedActualDocument as $key => $value) {
-            if (! $normalizedExpectedDocument->offsetExists($key)) {
-                $extraKeys[] = $key;
-            }
-        }
-
-        foreach ($extraKeys as $key) {
-            $normalizedActualDocument->offsetUnset($key);
-        }
-
-        $this->assertEquals(
-            Document::fromPHP($normalizedExpectedDocument)->toRelaxedExtendedJSON(),
-            Document::fromPHP($normalizedActualDocument)->toRelaxedExtendedJSON(),
-        );
+        (new DocumentsMatchConstraint($expectedDocument, true, true))->evaluate($actualDocument);
```

Comment:
Noted both parameters are to ignore extra keys. The purpose of this method is preserved.
https://github.com/mongodb/mongo-php-library/blob/70f8704ddc59fc25b085f8c98945f267506989dd/tests/SpecTests/DocumentsMatchConstraint.php#L69C47-L69C68